docs/src/components/ComponentPage/AndroidComponentPage.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow strict */
+
+import React from 'react';
+import BpkLink from 'bpk-component-link';
+
+import ComponentScreenshots from '../DocsPageBuilder/ComponentScreenshots';
+
+import ComponentPage from './ComponentPage';
+
+type Props = {
+  documentationId: string,
+  readme: string,
+  screenshots: [
+    {
+      id: string,
+      screenshots: {
+        altText: string,
+        height: number,
+        src: string,
+        subText: string,
+        title: string,
+        width: number,
+      },
+      title: string,
+    },
+  ],
+  usageTable: ?{
+    dos: Array<string>,
+    donts: Array<string>,
+  },
+};
+
+const AndroidComponentPage = (props: Props) => {
+  const { documentationId, screenshots, readme, usageTable } = props;
+  const screenshotsAsExamples = screenshots.map(screenshot => ({
+    id: screenshot.id,
+    title: screenshot.title,
+    content: <ComponentScreenshots screenshots={screenshot.screenshots} />,
+  }));
+  return (
+    <ComponentPage
+      usageTable={usageTable}
+      examples={screenshotsAsExamples}
+      readme={readme}
+      additionalContent={[
+        {
+          id: 'docs',
+          title: 'Class reference',
+          content: (
+            <BpkLink href={`/android/versions/latest/${documentationId}`} blank>
+              View on Backpack&apos;s Android documentation
+            </BpkLink>
+          ),
+        },
+      ]}
```

Comment:
If additional content was added by the consumer, shouldn't it be added here too?

Reply:
There is no additional content a consumer can add to this component. 

When we add the ability for pages to display more things (e.g. a link to view the repo on Jetpack) we'll add it as another prop which is then passed as part of the `additionalContent` array.

Reply:
Oh so `additionalContent` is strictly for the 4 page types to use? That seems overly restrictive. Would it make more sense to adopt a compositional approach, where the Pages accept an array of anything, and we create common elements to pass in?

Reply:
For example, `usageTable` isn't specific to Android pages, so why don't we just pass in a UsageTable node to `additionalContent`?

Reply:
> Oh so additionalContent is strictly for the 4 page types to use?
I don't really understand the question, because every prop on `ComponentPage` is for these pages to use.

> For example, usageTable isn't specific to Android pages, so why don't we just pass in a UsageTable node to additionalContent?

I did want to do this, where `ComponentPage` would just take `content` and then would only render it, as well as generate a table of contents. It's what I started out doing initially.

I changed to the route I'm proposing now because `usageTable` needs to be at the top of the page, and passing it into `additionalContent` would put it at the bottom. This is also why I asked on Slack about moving the usage table to the bottom, to enable this API.

Reply:
You could, however, make this argument that `readme` should be passed in as `additionalContent`. The downside would be that the logic to convert it to markdown would be duplicated in four places. What do you reckon?

Reply:
Ah right yeah that makes sense - but what about other sections in the future. Like if we want a `copy` section like Luna has, we'd have to add that to all 4 page types too?

Reply:
Yeah for README I guess that makes sense too. The duplicated logic could just be a method call, so seems fine to me

Reply:
I've not heard of Luna but I googled it and assume it's [this one](https://luna.sainsburys.co.uk/components/autocomplete), let me know if I've got the wrong one.

Yeah so in this scenario there's two ways to do it:

**First**
Add a `copy` prop to `ComponentPage`, which the four page types will pass down (like how we have done readme).

**Second**
Add a `copy` prop to each of the platform-specific component pages, and then pass it down as part of the additional content array.

I'd advocate for the first option. I feel like if it's a prop that can apply to all four platforms it should be centralised (like how readme is now), and when it's something specific to a particular platform it should be passed in as additional content (like how the links to docs are).

Here's a more true to life example based on our roadmap. We want to add links to view iOS components on Cocoapods, and we also want to add a link to view the source code of components on GitHub. 

Because the Cocoapods one is specific to iOS I would pass it as part of `additionalContent`, but for GitHub we know it applies to all platforms, so I'd add it as a prop to `ComponentPage`.

Reply:
For either of those examples, you could leave the Pages unmodified and consumers could use:

```
<AndroidPage ... additionalContent={[
  githubLink("/src/..."),
} />
```

That way we are composing the page from reusable sections, and whatever sections we wish to add in the future we can without turning this class into a monster

Reply:
Sorry for replying to this at such an unsociable time, but I've just come on my laptop to do something else and saw the comment (no need to reply until tomorrow).

I think you're right, but I also think maybe you've missed one aspect. Or I have 😅 I'll DM you tomorrow morning and we can talk it through on Zoom, will be easier than going back and forth on here

Reply:
Sounds great 👍